server/command.js

```diff
@@ -0,0 +1,243 @@
+var $ = function(str) {
+  // Parse the string to extract a reference.
+  // TODO: Support notations starting with @roomname and ~username.
+  var m = str.match(/^\s*(\$|me|~|here|@)(.*)\s*$/);
+  if (!m) {
+    return undefined;
+  }
+  var root = m[1];
+  var suffix = m[2];
+  if (root === '$') {
+    root = $;
+  } else if (root === '~' || root === 'me') {
+    root = user;
+  } else if (root === '@' || root === 'here') {
+    root = user.location;
+  } else {
+    throw Error("Can't happen.  Regex is too liberal.");
+  }
+  if (!suffix) {
+    return root;
+  }
+  // Clear no-longer needed local variables so they don't leak into the eval.
+  str = undefined;
+  m = undefined;
+  return eval('(root)' + suffix);
+};
+
+
+$.utils = {};
+$.utils.command = {};
+
+$.utils.command.prepositions = Object.create(null);
+$.utils.command.prepositions['with'] = 'with/using';
+$.utils.command.prepositions['using'] = 'with/using';
+$.utils.command.prepositions['at'] = 'at/to';
```

Comment:
> Why not initialise $.utils.command.prepositions and .prepositionsRegExp from an array literal?

This is code that will be thrown away once a minimal database is self-sustaining.  Such an initialisation function would be lost.

> Or omit the former, since it does not appear to be used.

Oops, good point, they are absolutely needed in (and were written for) $.utils.command.execute.  Added.

> And if we think it's likely that the list of prepositions will need to be modified in future 

I don't think we ever modified the preps in Moo Canada.  With the exception of when we attempted to translate everything to French.  But I'm not willing to write the entire database in an easily localizable form since that would mean harder to read code, slower execution, and vastly more work.  Times ten if one wants to think about Arabic and Hebrew.